### PR TITLE
support TZID= in schedule rrules

### PR DESCRIPTION
**Pull request by ryanpetrello**
this commit allows schedule `rrule` strings to include local timezone
information via `TZID=NNNNN`; occurrences are _generated_ in the local
time specified by the user (or UTC, if e.g., `DTSTART:YYYYMMDDTHHMMSSZ`)
while `Schedule.next_run`, `Schedule.dtstart`, and `Schedule.dtend` will be
stored in the UTC equivalent (i.e., the scheduler will still do math on
"what to run next" based on UTC datetimes).

in addition to this change, there is now a new API endpoint,
`/api/v2/schedules/preview/`, which takes an rrule and shows the next
10 occurrences in local and UTC time.

see: https://github.com/ansible/ansible-tower/issues/823
related: https://github.com/dateutil/dateutil/issues/614

### CODE REVIEW

**Comment by ryanpetrello** [comment]
This is very much a work in progress, but I wanted to get some eyes on my code so far - the last thing I need to do is write _lots_ of tests for edge cases.

**Comment by ryanpetrello** [comment]
Put on your thinking caps.

Here.  Be.  🐉 s

File: docs/schedules.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+`DTSTART` values provided to awx _must_ provide timezone information (they may
+not be naive dates).
+
+For UTC dates, `DTSTART` values should be denoted with the `Z` suffix:
+
+    DTSTART:20300115T120000Z
+
+Local timezones can be specified using the `TZID=` parameter:
+
+    DTSTART;TZID=America/New_York:20300115T120000Z
```
Comment (gconsidine) [inline docs/schedules.md]: `Z` should be omitted here since we're dealing with a non-UTC timezone, correct?

Comment (ryanpetrello) [reply docs/schedules.md]: Yep, good catch.

File: awx/main/tests/functional/models/test_schedule.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
+from awx.main.models import JobTemplate, Schedule
```
Comment (ryanpetrello) [inline awx/main/tests/functional/models/test_schedule.py]: Reviewers: please help me think of other wacky test cases to add here.

Comment (matburt) [reply awx/main/tests/functional/models/test_schedule.py]: See some of my `old` tests here: https://github.com/ansible/awx/blob/1.0.0/awx/main/tests/old/schedules.py

**Review by matburt — APPROVED** [review]

File: awx/main/tests/functional/models/test_schedule.py (hunk 2 above)
Comment (ryanpetrello) [reply awx/main/tests/functional/models/test_schedule.py]: Gonna replicate some of your "infinite recurrence" tests.

**Review by wwitzel3 — APPROVED** [review]

**Comment by wenottingham** [comment]
Some test cases, *as driven by the UI*:

Weekly, every Tuesday, TZ America/New_York, time 21:00:00
Monthly, on the first Sunday, TZ America/New_York, time 22:00:00
Yearly, on December 31, TZ America/New_York, time 23:00:00
etc.
Weekly, every Sunday, TZ America/New_York, time 02:30:00, start date 2018-03-11.

**Comment by wenottingham** [comment]
... any case where we're in non-DST and specify a start date and clock time that is post-DST-change. (or vice-versa)

File: awx/api/views.py [hunk 3]
```diff
@@ -608,6 +609,43 @@ class ScheduleDetail(RetrieveUpdateDestroyAPIView):
...
+    def post(self, request):
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            next_stamp = now()
+            schedule = []
+            gen = Schedule.rrulestr(serializer.validated_data['rrule']).xafter(next_stamp, count=20)
+
+            # loop across the entire generator and grab the first 10 events
+            for event in gen:
+                if len(schedule) >= 10:
```
Comment (ryanpetrello) [inline awx/api/views.py:628]: We need more than 10 events from the generator (20 should probably suffice) in case there's a few "imaginary dates", e.g., "the night of a DST switch between the hours of 2AM and 2:59:59AM"
